openapi-cli/src/test/java/io/ballerina/openapi/generators/openapi/RecordTests.java

```diff
@@ -70,6 +70,25 @@ public void testNullableFieldWithTypeReference() throws OpenApiConverterExceptio
         TestUtils.compareWithGeneratedFile(ballerinaFilePath, "record/nullable03.yaml");
     }
 
+    @Test(description = "When the record field has nullable field , which has record type reference data type")
+    public void testUnionTypeWithRecordType() throws OpenApiConverterException, IOException {
+        Path ballerinaFilePath = RES_DIR.resolve("record/union.bal");
+        TestUtils.compareWithGeneratedFile(ballerinaFilePath, "record/union.yaml");
+    }
+
+    @Test(description = "When the record field has nullable field , which has record type reference data type")
+    public void testUnionTypeWithPrimitive() throws OpenApiConverterException, IOException {
+        Path ballerinaFilePath = RES_DIR.resolve("record/union_with_primitive.bal");
+        TestUtils.compareWithGeneratedFile(ballerinaFilePath, "record/union_with_primitive.yaml");
+    }
+
+    @Test(description = "When the record field has nullable field , which has record type reference data type",
+            enabled = false)
```

Comment:
Is this disabled intentionally?

Reply:
yeah due to that ballerina lang known issue this is disabled.